directory/directory_test.go

```diff
@@ -145,7 +145,7 @@ func TestGetPutSignatures(t *testing.T) {
 	src, err := ref.NewImageSource(nil, nil)
 	require.NoError(t, err)
 	defer src.Close()
-	sigs, err := src.GetSignatures()
+	sigs, err := src.GetSignatures(nil)
```

Comment:
I believe we want to pass `context.TODO` here even if this is a test (or context.Background better)